src/anon/middleware.js

```diff
@@ -19,7 +19,31 @@ function anonymousMiddleware(req, res, next){
 		|| req.url;
 	res.locals.navigationModel = new NavigationModel(res.locals.flags, res.locals.anon.userIsAnonymous, currentUrl);
 	res.vary('FT-Anonymous-User');
-	next();
+
+	if (res.locals.flags.brexitDiscount && res.locals.flags.brexitDiscountType) {
+		getBarrierData(req).then(function (response) {
+			res.brexitOfferBasePrice = response.viewData.subscriptionOptions.STANDARD.price.weekly;
+			next();
+		});
+	}
+	else {
+		next();
+	}
+}
+
+function getBarrierData (req) {
+
+	const headers = {
+		'Content-Classification': req.query['ft-content-classification'] || req.get('FT-Content-Classification') || 'CONDITIONAL_STANDARD',
+		'Country-Code': req.query['country-code'] || req.get('country-code') || 'GBR',
+	};
+
+	if (headers['Content-Classification'] === '-') {
+		headers['Content-Classification'] = 'CONDITIONAL_STANDARD';
+	}
+
+	return fetch('https://barrier-app.memb.ft.com/memb/barrier/v1/barrier-data', { headers: headers, timeout: 2000 })
+		.then(res => res.json())
```

Comment:
doesn't appear to be any error handling for this request if it fails or get a bad error status